admin_apps/shared_utils.py

```diff
@@ -65,7 +77,7 @@ def init_session_states() -> None:
         st.session_state.semantic_model = semantic_model_pb2.SemanticModel()
     # validated stores the status if the generated yaml has ever been validated.
     if "validated" not in st.session_state:
-        st.session_state.validated = False
+        st.session_state.validated = None
```

Comment:
Using `None` to mean "we haven't validated yet", so we can reserve `False` for "invalid"